app/models/miq_set.rb

```diff
@@ -0,0 +1,66 @@
+class MiqSet < ApplicationRecord
+  # self.abstract_class = true
+  self.inheritance_column = :set_type
+
+  include UuidMixin
+
+  acts_as_miq_taggable
+
+  serialize :set_data
+
+  validates :name, :presence => true, :uniqueness_when_changed => {:scope => [:set_type, :userid, :group_id], :if => proc { |c| c.class.in_my_region.exists?(:name => c.name) }}
+  validates :description, :presence => true
+
+  has_many :miq_set_memberships, :dependent => :delete_all
+  belongs_to :owner, :polymorphic => true
+
+  attr_writer :model_class
+
+  def self.model_class
+    @model_class ||= name[0..-4].constantize
+  end
+
+  def self.model_table_name
+    @model_table_name ||= model_class.table_name
+  end
+
+  def members
+    miq_set_memberships.includes(:member).collect(&:member)
+  end
+  alias miq_sets members
+  alias children members
+
+  def members=(*members)
+    transaction do
+      remove_all_members
+      add_members(*members)
+    end
+  end
+  alias replace_children members=
+
+  def remove_member(member)
+    miq_set_memberships.where(:member => member).delete_all
+  end
+  alias remove_child remove_member
+
+  def remove_all_members
+    miq_set_memberships.delete_all
+  end
+  alias remove_all_children remove_all_members
+
+  def add_members(*members)
```

Comment:
this does have unique membership
Can we do that with a traditional association out of the box?